File: 홍연일/4주차/1806.py
def main():
    n, s = map(int, input().split())
    a = list(map(int, input().split()))
    
    left = 0
    right = 0
    current_sum = 0
    min_length = n + 1
    
    while right < n:
        current_sum += a[right]
        
        while current_sum >= s:
            min_length = min(min_length, right - left + 1)
            current_sum -= a[left]
            left += 1
        
        right += 1
    
    if min_length > n:
        print(0)
    else:
        print(min_length)

if __name__ == "__main__":
    main()

File: 한성준/week4/two_pointer/1806.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int N, S;
    cin >> N >> S;
    
    vector<int> numbers(N);
    for (int i = 0; i < N; i++) {
        cin >> numbers[i];
    }
    
    int left = 0, right = 0;
    int sum = 0;
    int minLength = INT_MAX;  // 최소 길이
    
    while (right < N) {
        // 오른쪽 포인터 이동하여 합 증가
        sum += numbers[right++];
        
        // 합이 S 이상이면 왼쪽 포인터 이동
        while (sum >= S) {
            minLength = min(minLength, right - left);  // 현재 길이 갱신
            sum -= numbers[left++];  // 왼쪽 값 제거
        }
    }
    
    if (minLength == INT_MAX) {
        cout << 0 << '\n';  // 합이 S 이상인 부분합이 없는 경우
    } else {
        cout << minLength << '\n';
    }
    
    return 0;
}
